O/OpenFOAM_com/build_tarballs.jl

```diff
@@ -0,0 +1,577 @@
+    ExecutableProduct("surfaceGenerateBoundingBox", :surfaceGenerateBoundingBox),
+    ExecutableProduct("surfaceHookUp", :surfaceHookUp),
+    ExecutableProduct("surfaceInertia", :surfaceInertia),
+    ExecutableProduct("surfaceInflate", :surfaceInflate),
+    ExecutableProduct("surfaceLambdaMuSmooth", :surfaceLambdaMuSmooth),
+    ExecutableProduct("surfaceMeshConvert", :surfaceMeshConvert),
+    ExecutableProduct("surfaceMeshExport", :surfaceMeshExport),
+    ExecutableProduct("surfaceMeshExtract", :surfaceMeshExtract),
+    ExecutableProduct("surfaceMeshImport", :surfaceMeshImport),
+    ExecutableProduct("surfaceMeshInfo", :surfaceMeshInfo),
+    ExecutableProduct("surfaceOrient", :surfaceOrient),
+    ExecutableProduct("surfacePatch", :surfacePatch),
+    ExecutableProduct("surfacePointMerge", :surfacePointMerge),
+    ExecutableProduct("surfaceRedistributePar", :surfaceRedistributePar),
+    ExecutableProduct("surfaceRefineRedGreen", :surfaceRefineRedGreen),
+    ExecutableProduct("surfaceSplitByPatch", :surfaceSplitByPatch),
+    ExecutableProduct("surfaceSplitByTopology", :surfaceSplitByTopology),
+    ExecutableProduct("surfaceSplitNonManifolds", :surfaceSplitNonManifolds),
+    ExecutableProduct("surfaceSubset", :surfaceSubset),
+    ExecutableProduct("surfaceToFMS", :surfaceToFMS),
+    ExecutableProduct("surfaceToPatch", :surfaceToPatch),
+    ExecutableProduct("surfaceTransformPoints", :surfaceTransformPoints),
+    ExecutableProduct("surfactantFoam", :surfactantFoam),
+    ExecutableProduct("temporalInterpolate", :temporalInterpolate),
+    ExecutableProduct("tetgenToFoam", :tetgenToFoam),
+    ExecutableProduct("tetMesh", :tetMesh),
+    ExecutableProduct("thermoFoam", :thermoFoam),
+    ExecutableProduct("topoSet", :topoSet),
+    ExecutableProduct("transformPoints", :transformPoints),
+    ExecutableProduct("twoLiquidMixingFoam", :twoLiquidMixingFoam),
+    ExecutableProduct("twoPhaseEulerFoam", :twoPhaseEulerFoam),
+    ExecutableProduct("uncoupledKinematicParcelDyMFoam", :uncoupledKinematicParcelDyMFoam),
+    ExecutableProduct("uncoupledKinematicParcelFoam", :uncoupledKinematicParcelFoam),
+    ExecutableProduct("viewFactorsGen", :viewFactorsGen),
+    ExecutableProduct("vtkUnstructuredToFoam", :vtkUnstructuredToFoam),
+    ExecutableProduct("wallFunctionTable", :wallFunctionTable),
+    ExecutableProduct("writeMeshObj", :writeMeshObj),
+    ExecutableProduct("writeMorpherCPs", :writeMorpherCPs),
+    ExecutableProduct("XiDyMFoam", :XiDyMFoam),
+    ExecutableProduct("XiEngineFoam", :XiEngineFoam),
+    ExecutableProduct("XiFoam", :XiFoam),
+    ExecutableProduct("zipUpMesh", :zipUpMesh),
+    
+    FileProduct("share/openfoam/etc", :openfoam_etc), 
+    FileProduct("share/openfoam/bin", :openfoam_bin)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="flex_jll", uuid="48a596b8-cc7a-5e48-b182-65f75e8595d0"))
```

Comment:
How is flex used here?  It's usually a `HostBuildDependency`, because it's a tool used only during the build, and has to run on the host, does openfoam does something special with it at runtime?

Reply:
When I change it to `HostBuildDependency` I get the following error:
`[23:16:27] /workspace/srcdir/openfoam/build/linux64GccDPInt32Opt/src/fileFormats/stl/STLAsciiParseFlex.L.C:319:10: fatal error: FlexLexer.h: No such file or directory
[23:16:27]   319 | 
[23:16:27]       |          ^            
[23:16:27] compilation terminated.`
When it is a normal `Dependency` it works fine.